gke-usage-metering/src/gkeUsageMetering.ts

```diff
@@ -141,7 +141,10 @@ namespace gkeUsageMetering {
         .setText('The resource usage dataset ID is not specified')
         .throwException();
     }
-    if (fullResourceUsageDataset.indexOf('.') < 0) {
+    // GCP project name can contain letters, numbers, single quotes, 
+    // hyphens, spaces or exclamation points; BigQuery dataset IDs can contain
+    // letters, numbers and underscores
+    if (!fullResourceUsageDataset.match(/^[a-zA-Z-\d' !]+\.[a-zA-Z_\d]+$/)) {
```

Comment:
```
Project IDs must start with a lowercase letter and can have lowercase ASCII letters, digits or hyphens.
```

So I believe the first part of the regex should be `[a-z][a-z-\d]+`.

Reply:
Done